File: backend/src/main/java/com/campus/Campus_Connect/features/metadata/courses/CourseMapper.java
package com.campus.Campus_Connect.features.metadata.courses;

import com.campus.Campus_Connect.features.metadata.courses.dto.CourseResponse;
import com.campus.Campus_Connect.features.metadata.courses.dto.DetailedCourseResponse;
import com.campus.Campus_Connect.features.metadata.courses.entity.Course;
import org.springframework.stereotype.Component;

@Component
public class CourseMapper {

    public CourseResponse toCourseResponse(Course course) {

        return CourseResponse.builder()
                .courseId(course.getId())
                .degree(course.getDegree())
                .programname(course.getProgramName())
                .courseCode(course.getCourseCode())
                .build();
    }

    public DetailedCourseResponse toDetailedCourseResponse(Course course) {

        return DetailedCourseResponse.builder()
                .courseId(course.getId())
                .degree(course.getDegree())
                .programName(course.getProgramName())
                .courseCode(course.getCourseCode())
                .degreeLevel(course.getDegreeLevel())
                .durationYears(course.getDurationYears())
                .build();
    }
}

File: android-app/app/src/main/java/com/example/campusconnect/feature/metadata/courses/local/CourseMapper.kt
package com.example.campusconnect.feature.metadata.courses.local

import com.example.campusconnect.feature.metadata.courses.Course
import com.example.campusconnect.feature.metadata.courses.remote.CourseResponse

fun CourseEntity.toCourse(): Course {
    return Course(
        courseId = courseId,
        degree = degree,
        programName = programName,
        courseCode = courseCode,
        degreeLevel = degreeLevel,
        durationYears = durationYears
    )

}

fun Course.toEntity(): CourseEntity {
    return CourseEntity(
        courseId = courseId,
        degree = degree,
        programName = programName,
        courseCode = courseCode,
        degreeLevel = degreeLevel,
        durationYears = durationYears
    )
}

fun CourseResponse.toCourse(): Course {
    return Course(
        courseId = courseId,
        degree = degree,
        programName = programName,
        courseCode = courseCode,
        degreeLevel = degreeLevel,
        durationYears = durationYears
    )
}

